tools/istio-iptables/main.go

```diff
@@ -0,0 +1,516 @@
+	if proxyUID == "" {
+		usr, err := ext.LookupUser()
+		var userID string
+		// Default to the UID of ENVOY_USER and root
+		if err != nil {
+			userID = "1337"
+		} else {
+			userID = usr.Uid
+		}
+		// If ENVOY_UID is not explicitly defined (as it would be in k8s env), we add root to the list,
+		// for ca agent.
+		proxyUID = userID + ",0"
+	}
+
+	// for TPROXY as its uid and gid are same
+	if proxyGID == "" {
+		proxyGID = proxyUID
+	}
+
+	podIP, err := ext.GetLocalIP()
+	if err != nil {
+		panic(err)
+	}
+	// Check if pod's ip is ipv4 or ipv6, in case of ipv6 set variable
+	// to program ip6tablesOrFail
+	if podIP.To4() == nil {
+		enableInboundIPv6s = podIP
+	}
+	//
+	// Since OUTBOUND_IP_RANGES_EXCLUDE could carry ipv4 and ipv6 ranges
+	// need to split them in different arrays one for ipv4 and one for ipv6
+	// in order to not to fail
+	ipv4RangesExclude, ipv6RangesExclude, err := separateV4V6(outboundIPRangesExclude)
+	if err != nil {
+		panic(err)
+	}
+	if ipv4RangesExclude.IsWildcard {
+		panic("Invalid value for OUTBOUND_IP_RANGES_EXCLUDE")
+	}
+
+	ipv4RangesInclude, ipv6RangesInclude, err := separateV4V6(outboundIPRangesInclude)
+	if err != nil {
+		panic(err)
+	}
+
+	// Remove the old chains, to generate new configs.
+	ext.RunQuietlyAndIgnore(dep.IPTABLES, "-t", "nat", "-D", "PREROUTING", "-p", "tcp", "-j", "ISTIO_INBOUND")
+	ext.RunQuietlyAndIgnore(dep.IPTABLES, "-t", "mangle", "-D", "PREROUTING", "-p", "tcp", "-j", "ISTIO_INBOUND")
+	ext.RunQuietlyAndIgnore(dep.IPTABLES, "-t", "nat", "-D", "OUTPUT", "-p", "tcp", "-j", "ISTIO_OUTPUT")
+	// Flush and delete the istio chains.
+	ext.RunQuietlyAndIgnore(dep.IPTABLES, "-t", "nat", "-F", "ISTIO_OUTPUT")
+	ext.RunQuietlyAndIgnore(dep.IPTABLES, "-t", "nat", "-X", "ISTIO_OUTPUT")
+	ext.RunQuietlyAndIgnore(dep.IPTABLES, "-t", "nat", "-F", "ISTIO_INBOUND")
+	ext.RunQuietlyAndIgnore(dep.IPTABLES, "-t", "nat", "-X", "ISTIO_INBOUND")
+	ext.RunQuietlyAndIgnore(dep.IPTABLES, "-t", "mangle", "-F", "ISTIO_INBOUND")
+	ext.RunQuietlyAndIgnore(dep.IPTABLES, "-t", "mangle", "-X", "ISTIO_INBOUND")
+	ext.RunQuietlyAndIgnore(dep.IPTABLES, "-t", "mangle", "-F", "ISTIO_DIVERT")
+	ext.RunQuietlyAndIgnore(dep.IPTABLES, "-t", "mangle", "-X", "ISTIO_DIVERT")
+	ext.RunQuietlyAndIgnore(dep.IPTABLES, "-t", "mangle", "-F", "ISTIO_TPROXY")
+	ext.RunQuietlyAndIgnore(dep.IPTABLES, "-t", "mangle", "-X", "ISTIO_TPROXY")
+	// Must be last, the others refer to it
+	ext.RunQuietlyAndIgnore(dep.IPTABLES, "-t", "nat", "-F", "ISTIO_REDIRECT")
+	ext.RunQuietlyAndIgnore(dep.IPTABLES, "-t", "nat", "-X", "ISTIO_REDIRECT")
+	ext.RunQuietlyAndIgnore(dep.IPTABLES, "-t", "nat", "-F", "ISTIO_IN_REDIRECT")
+	ext.RunQuietlyAndIgnore(dep.IPTABLES, "-t", "nat", "-X", "ISTIO_IN_REDIRECT")
+
+	if len(flagSet.Args()) > 0 && flagSet.Arg(0) == "clean" {
+		fmt.Println("Only cleaning, no new rules added")
+		return
+	}
+	// Dump out our environment for debugging purposes.
+	fmt.Println("Environment:")
```

Comment:
use Istio's built in logger please.

Reply:
We are also in favour to replace this, but:

Currently, the tests to keep `istio-iptables.sh` and this go implementation in sync are based on the output of both implementations. 

Therefore, it's not possible to replace this with a logger, because the logger might prefix the message by a timestamp.